provider_test.go

```diff
@@ -33,3 +33,15 @@ func testAccPreCheck(t *testing.T) {
 		t.Fatal("ELASTICSEARCH_URL must be set for acceptance tests")
 	}
 }
+
+func testAccPreCheckXpack(t *testing.T) {
+	if v := os.Getenv("ELASTICSEARCH_URL"); v == "" {
```

Comment:
This could be replaced by `testAccPreCheck(t)`

Reply:
I added one with more check because this feature requires XPack which also requires authentication. This PreCheck makes sure the username and password are also set.

Reply:
I mean ```
if v := os.Getenv("ELASTICSEARCH_URL"); v == "" {
		t.Fatal("ELASTICSEARCH_URL must be set for acceptance tests")
	}
```
is replicated
You can call `testAccPreCheck` in `testAccPreCheckXpack`

Reply:
Oh right, good point!